src/shell.js

```diff
@@ -233,11 +233,12 @@ if (ENVIRONMENT_IS_SHELL) {
     };
   }
 
-  if (typeof console !== 'undefined') {
+  if (typeof console === 'undefined') {
     // Support odd shell environments that lack console.* but have other stuff.
-    console = {
+    var console = {
```

Comment:
I think you want to drop the var since console is actually a global global thing.